### ISSUE TITLE
Potential bug in datasets suggested by uniform score

### ISSUE THREAD

**Issue by KennethEnevoldsen** [embeddings-benchmark/mteb#1172]
![Screenshot 2024-08-21 at 14 05 10](https://github.com/user-attachments/assets/8a8053c8-e9dd-4c5b-aca8-fbe8dfa3e02c)

The datasets are: 'Diversity1LegalBenchClassification', 'Diversity2LegalBenchClassification', 'IFlyTek', 'TNews' and they score is always the same across a variety of models.

**Comment by AlexeyVatolin** [comment]
In the datasets `Diversity1LegalBenchClassification` and `Diversity2LegalBenchClassification` we have 6 examples in training and only one with a positive class. I don't think this is enough to train the model. Maybe we need to remove these datasets?
In `IFlyTek` and `TNews` we have the `validation` and `test` sets. In the `test` set the label column is -1 for all examples, but in the validation set there is no such problem. I propose to use the validation set to calculate the metric

**Comment by KennethEnevoldsen** [comment]
> In the datasets Diversity1LegalBenchClassification and Diversity2LegalBenchClassification we have 6 examples in training and only one with a positive class. I don't think this is enough to train the model. Maybe we need to remove these datasets?

That is totally fine to keep for me. In that case, they are just highly challenging and might encourage new embedding approaches.

> In IFlyTek and TNews we have the validation and test sets. In the test set the label column is -1 for all examples, but in the validation set there is no such problem. I propose to use the validation set to calculate the metric

I agree with that suggestion and have added a PR for it #1190

**Comment by AlexeyVatolin** [comment]
I agree with you. At this point, I think we can consider this issue resolved and close the issue
